queries/vue/injections.scm

```diff
@@ -1,26 +1,12 @@
 ((style_element
-  (raw_text) @css))
-
-(
-  (style_element
-    (start_tag
-      (attribute
-        (quoted_attribute_value (attribute_value) @_lang)))
-    (raw_text) @scss)
-  (#match? @_lang "(scss|postcss|less)")
-)
+  (start_tag (attribute (quoted_attribute_value (attribute_value) @_lang))?)
+  (raw_text) @content)
+ (#inject! @_lang "css"))
```

Comment:
Do the changes in this file have the below logic for Vue?
1. > When you encounter a `<style lang="scss">` tag, inject scss
2. > When you encounter a `<style>` (no lang attribute), inject css

I see this PR removes

```
 (raw_text) @scss)
  (#match? @_lang "(scss|postcss|less)")
```
Doesn't it mean scss is never injected?

When I open your branch on a clean config (just nvim-treesitter) I get:
![image](https://user-images.githubusercontent.com/5359825/115974002-c0c8f580-a559-11eb-9d29-37d11e73394d.png)  
(the `$variableandor:` is scss specific syntax) 

The fix here would be to implement what is mentioned here: https://github.com/nvim-treesitter/nvim-treesitter/issues/1209#issuecomment-823878890

So if lang="scss", inject only scss, but if no lang attribute, inject css

I'd also keep the "inject scss for scss,postcss,less" (  (#match? @_lang "(scss|postcss|less)")) line for now

Reply:
Is the situation with `<script lang="ts">` not similar? ie. typescript not injected

Reply:
This change injects css if there isn't a lang tag, otherwise injects the language from the lang tag, if postcss or less is given, it will inject scss (they are defined as aliases).

In the adobe screenshot scss is being injected. The variables are white bc they are capture like so in https://github.com/stsewd/nvim-treesitter/blob/injection-regex/queries/scss/highlights.scm (and looks like there are some missing captures for `$` and `:`, you can check this by opening a `test.scss` file with that content.

Reply:
looks like `:` is already captured, and the parser doesn't distinguish `$` from the var name.

Reply:
:man_facepalming: it's all in the PR code, the aliases make sense. Sorry

tbh my knowledge about the project is rather limited and I don't want to keep bugging you with my potentially missed shots, so I'll just dump what I came across so you or the maintainers could also confirm it if it appears on their radar:

* if I check out to your branch and do `nvim index.vue`, I get
    `Error detected while processing FileType Autocommands for "*":
E5108: Error executing lua /usr/share/nvim/runtime/lua/vim/treesitter/lan
guagetree.lua:337: table index is nil`  
This doesn't happen if I just open `nvim` and then navigate to the file using telescope etc. If I revert back to master this issue disappears (this is reproduceable with a clean config like https://github.com/3nuc/tree-sitter-vue-issue-10/tree/stsewd, repro instructions are in issue 1209). This may be just some weird cached stuff on my side so do let me know if it works fine on your side. So perhaps this is why I'm running into the injection problems.

Reply:
Nah, feel free to keep posting all errors you find. I wasn't able to replicate this with  https://github.com/3nuc/tree-sitter-vue-issue-10/blob/stsewd/index.vue, can you try modifying the file like adding a new line and saving it and opening it again and see if stills happens?

Reply:
Yeah I still get this even if I change the file, save and repoen. Do you mean you couldn't replicate it by running index.vue on your config or that you cloned that repo with --recurse-submodules and ran ./run.sh from there?

Reply:
I wasn't able to replicate with ./run.sh in that branch, yeah.

Reply:
Let's just write it off as something wrong with my setup for now. If I manage to reproducce it on something squeaky clean like an empty VM with just neovim I'll let you know here.

Reply:
FWIW you could use Nix and try to run Neovim inside a pure shell, it's easier than creating a VM in my opinion